File: ui/src/app/model/location.ts
export class Location {
    id: number;
    code: string;
    name: string;

    constructor(id: number = null, code: string = null, name: string = null) {
        this.id = id;
        this.code = code;
        this.name = name;
    }

    getId(): number {
        return this.id;
    }
    getCode(): string {
        return this.code;
    }
    getName(): string {
        return this.name;
    }

    equals(other: Location): boolean {
        return this.getCode() === other.getCode() &&
        this.getName() === other.getName();
    }
}

File: src/main/java/com/mni/model/location/Location.java
package com.mni.model.location;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Location {
    @Id
    @GeneratedValue
    private Long id;

    @Column(unique = true)
    private String code;

    @Column(unique = true)
    private String name;


    public Location() {}

    public Location(Long id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Location{" +
                "id=" + id +
                ", code='" + code + '\'' +
                ", name='" + name + '\'' +
                '}';
    }
}
